alibi/explainers/partial_dependence.py

```diff
@@ -874,46 +1138,114 @@ def _sample_ice(ice_values: np.ndarray, n_ice: Union[Literal['all'], int, List[i
+
+    Parameters
+    ----------
+    exp, n_ice, center
+        See :py:meth:`alibi.explainers.partial_dependence.plot_pd` method.
+    pd_values
+        Array of ice_values of dimension `V` (i.e. `(V, )`), where V is the number of feature values where
+        the PD is computed.
+    ice_values
+        Array of ice_values of dimension `N x V`, where `N` is the number of instances in the reference dataset,
+        and `V` is the number of feature values where the PD is computed.
+
+    Returns
+    -------
+    Tuple containing the processed `pd_values` and `ice_values`.
+    """
+    # pdp processing
+    if exp.meta['params']['kind'] == Kind.BOTH and center:
+        pd_values = pd_values - pd_values[0]  # type: ignore[index]
+
+    # ice processing
+    if exp.meta['params']['kind'] in [Kind.INDIVIDUAL, Kind.BOTH]:
+        # sample ice values for visualization purposes
+        ice_values = _sample_ice(ice_values=ice_values, n_ice=n_ice)  # type: ignore[arg-type]
+
+        # center ice values if necessary
+        if center:
+            ice_values = ice_values - ice_values[0:1, :]
+
+    return pd_values, ice_values
+
+
+def _compute_pd_limits(exp: Explanation,
+                       kind: Literal['average', 'individual', 'both'] = 'average',
+                       pd_values: Optional[np.ndarray] = None,
+                       ice_values: Optional[np.ndarray] = None,
+                       padding_proc: float = 0.1) -> Tuple[float, float]:
+    """
+    Computes the minimum and maximum y-limits for all the one-way PD plots
+
+    Parameters
+    ----------
+    exp, pd_values, ice_values
+        See :py:meth:`alibi.explainers.partial_dependence.plot_pd` method.
+    kind
+        See :py:meth:`alibi.explainers.partial_dependence.explain` method.
+    padding_proc
+        Padding percentage.
+
+    Returns
+    -------
+    Tuple containing the minimum and maximum y-limits.
+    """
+    values = pd_values if kind == Kind.AVERAGE else ice_values
+    min_val, max_val = values.min(), values.max()  # type: ignore[union-attr]
+    padding = padding_proc * (max_val - min_val)
+    return min_val - padding, max_val + padding
+
+
 # No type check due to the generic explanation object
 @no_type_check
 def _plot_one_pd_num(exp: Explanation,
                      feature: int,
                      target_idx: int,
-                     center: bool = True,
-                     n_ice: Union[Literal['all'], int, List[int]] = 'all',
+                     center: bool = False,
+                     pd_limits: Optional[Tuple[float, float]] = None,
+                     n_ice: Union[Literal['all'], int, List[int]] = 100,
                      ax: Optional['plt.Axes'] = None,
                      pd_num_kw: Optional[dict] = None,
-                     ice_num_kw: Optional[dict] = None) -> 'plt.Axes':
+                     ice_num_kw: Optional[dict] = None) -> Tuple['plt.Axes', Optional[Tuple[float, float]]]:
```

Comment:
Worth updating the docstrings for all these internal functions that now return more things.